base_layer/wallet/src/output_manager_service/service.rs

```diff
@@ -1888,97 +1891,159 @@ where
-                                    "Attempt to add scanned output {} that already exists. Ignoring the output.",
-                                    output_hex
-                                );
-                            },
-                            Err(err) => {
-                                return Err(err.into());
+        let scanned_outputs = outputs
+            .into_iter()
+            .filter_map(|output| {
+                // TODO: use MultiKey
+                match output.script.as_slice() {
+                    // ----------------------------------------------------------------------------
+                    // simple one-sided address
+                    [Opcode::PushPubKey(spending_key_public)] => {
+                        let keys = match self.resources.db.get_all_known_one_sided_payment_scripts() {
+                            Ok(keys) => keys,
+                            Err(_) => return None,
+                        };
+
+                        match keys
+                            .into_iter()
+                            .find(|x| PublicKey::from_secret_key(&x.private_key) == **spending_key_public)
+                        {
+                            None => None,
+                            Some(x) => {
+                                match PrivateKey::from_bytes(
+                                    CommsPublicKey::shared_secret(&x.private_key, &output.sender_offset_public_key)
+                                        .as_bytes(),
+                                ) {
+                                    Ok(spend_key) => Some((output.clone(), x.private_key.clone(), spend_key)),
+                                    Err(_) => None,
+                                }
                             },
                         }
-                        trace!(
-                            target: LOG_TARGET,
-                            "One-sided payment Output {} with value {} recovered",
-                            output_hex,
-                            committed_value,
-                        );
+                    },
+
+                    // ----------------------------------------------------------------------------
+                    // one-sided stealth address
+                    // NOTE: Extracting the nonce R and a spending (public aka scan_key) key from the script
+                    // NOTE: [RFC 203 on Stealth Addresses](https://rfc.tari.com/RFC-0203_StealthAddresses.html)
+                    [Opcode::PushPubKey(nonce), Opcode::Drop, Opcode::PushPubKey(provided_spending_public)] => {
+                        let secret_key = self.node_identity.secret_key().clone();
+                        let public_key = self.node_identity.public_key();
+
+                        // computing shared secret
+                        let c = RistrettoSecretKey::from_bytes(
+                            DomainSeparatedHasher::<Blake256, GenericHashDomain>::new("stealth address")
```

Comment:
The [hashing API](https://github.com/tari-project/tari-crypto/blob/main/src/hashing.rs) suggests a descriptive reverse-DNS structure for domain separation labels.

Reply:
we've changed label to `com.tari.stealth_address`, in both PRs; is that what you meant?